ir/src/tests/mod.rs

```diff
@@ -379,6 +379,126 @@ fn transition_constraints_using_parens() {
     assert!(result.is_ok());
 }
 
+#[test]
+fn random_values_fixed_list() {
+    let source = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [16]
+    boundary_constraints:
+        enf c.first = $rand[10] * 2
+        enf c.last = 1
+    integrity_constraints:
+        enf c' = $rand[3] + 1";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn random_values_custom_name() {
+    let source = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        alphas: [16]
+    boundary_constraints:
+        enf c.first = $alphas[10] * 2
+        enf c.last = 1
+    integrity_constraints:
+        enf c' = $alphas[3] + 1";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn random_values_ident_vector() {
```

Comment:
nit: Should this be renamed to `random_values_named_access` or something similar?